### PR TITLE
Add dht.rebind()

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -101,8 +100,15 @@ class DHT extends EventEmitter {
     if (buf.byteLength > 1) this.io.onmessage(socket, buf, rinfo)
   }
 
-  bind () {
-    return this.io.bind()
```
Comment (LuKks) [inline index.js:105]: Removing due not being useful for users, and it's automatically called when bootstrapping

Then we add docs for `.socket` and we fully close https://github.com/mafintosh/dht-rpc/issues/42

Comment (mafintosh) [reply index.js:105]: dont disagree, but thats a sep PR

File: index.js [hunk 2]
```diff
@@ -366,10 +372,15 @@ class DHT extends EventEmitter {
...
-      this.emit('ephemeral')
+    if (this.adaptive) {
+      this.firewalled = true
+      this.io.firewalled = true
+
+      if (!this.ephemeral) {
+        this.ephemeral = true
+        this.io.ephemeral = true
+        this.emit('ephemeral')
+      }
```
Comment (mafintosh) [inline index.js:383]: this change is technically unrelated, but we use warmup for rebind so ...

File: lib/errors.js [hunk 3]
```diff
@@ -7,6 +7,12 @@ exports.createTimeoutError = () => {
   return timeoutErr
 }
 
+exports.createRebindError = () => {
+  const reboundErr = new Error('Request socket rebound')
+  reboundErr.code = 'EREBOUND'
+  return reboundErr
+}
+
```
Comment (kasperisager) [inline lib/errors.js]: For future consideration: Should we use a proper `Error` subclass with error constructor functions for these, same as https://github.com/holepunchto/hypercore-errors?

Comment (mafintosh) [reply lib/errors.js]: Yea! Can you PR that after I merge?

**Pull request merged**